src/dictionary.c

```diff
@@ -0,0 +1,103 @@
+
+/*
+ * A dictionary data structure based on tries
+ *
+ * See dictionary.h for function documentation.
+ */
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include <assert.h>
+#include <unistd.h>
+#include "utils.h"
+#include "dictionary.h"
+
+/* See dictionary.h */
+dict_t* dict_new() {
+
+    dict_t *d;
+    int rc;
+
+    d = malloc(sizeof(dict_t));
+
+    if (d == NULL) {
+        error("Could not allocate memory");
+        return NULL;
+    }
+
+    rc = dict_init(d);
+    if (rc != 0) {
```

Comment:
Should be `EXIT_SUCCESS`/`EXIT_FAILURE` as appropriate

Reply:
The use of 1 and 0 here is inconsistent with `trie_new()`. This must be consistent.